src/bench/wallet_balance.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2012-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <interfaces/chain.h>
+#include <key_io.h>
+#include <optional.h>
+#include <test/util.h>
+#include <validationinterface.h>
+#include <wallet/wallet.h>
+
+struct WalletTestingSetup {
```

Comment:
This seems fine, but in the future it might be better not have two different structs called `WalletTestingSetup` with different implementations. (The other one is https://github.com/bitcoin/bitcoin/blob/master/src/wallet/test/wallet_test_fixture.h), because if the other `WalletTestingSetup` struct is ever used in a benchmark, it might result in weird link errors or even runtime errors.

I think if you just took the `handleNotifications` method below and moved it to the `CWallet` class as a public method, this struct would be simpler and no longer need to be declared a `CWallet` friend, and it could move to a private namespace to avoid conflicting with the other struct.